pybop/models/base_model.py

```diff
@@ -983,6 +1011,15 @@ def spatial_methods(self):
     def solver(self):
         return self._solver
 
+    @property
+    def calculate_sensitivities(self):
+        return self._calculate_sensitivities
+
     @solver.setter
     def solver(self, solver):
-        self._solver = solver.copy() if solver is not None else None
+        if isinstance(solver, pybamm.solvers.idaklu_jax.IDAKLUJax):
```

Comment:
In general, the user will receive an error within the `Problem.evaluateS1` method when trying to access the sensitivities from the solution: https://github.com/pybop-team/PyBOP/blob/424699be42ec0cf790a445ba208e0775a5a1317b/pybop/problems/fitting_problem.py#L211

Do you think we should cache a list of PyBaMM solvers that support sensitivities to compare against? This could go stale over time, although it wouldn't be much work to update given the small set of available solvers.

From the perspective of this PR, the jaxified IDAKLU supports `calculate_sensitivites`, so we should be fine.